lib/propmangle.js

```diff
@@ -202,9 +207,21 @@ function mangle_properties(ast, options) {
 
         var mangled = cache.props.get(name);
         if (!mangled) {
-            do {
-                mangled = base54(++cache.cname);
-            } while (!can_mangle(mangled));
+            if (debug) {
+                // debug mode: use a prefix and suffix to preserve readability, e.g. o.foo -> o._$foo$NNN_.
+                var debug_mangled = "_$" + name + "$" + debug_cache_name + "_";
+                
+                if (can_mangle(debug_mangled))
+                    mangled = debug_mangled;
+                else
+                    mangled = name;
```

Comment:
This name collision doesn't seem related to debug mangling, because ordinary mangling can have the same collision. The problem is quoted properties are added to the ignore list, but `can_mangle` does not check the ignore list, only `should_mangle` does. (I am pretty confused about what the difference between these functions is anyway.)

For example compile this with --mangle-props=2 without any debug mode:

```
var x = {};
var y = {};
y.foo = "bar";
x["a"] = y.foo;
console.log(x["a"]);
```

It generates a collision by outputting `x["a"] = y.a;`

I think this should be treated as a separate issue, or we could just ignore it?